ipa/srs_precomp_test.go

```diff
@@ -2,46 +2,84 @@ package ipa
 
 import (
 	"bytes"
+	"fmt"
 	"testing"
 
+	"github.com/crate-crypto/go-ipa/bandersnatch/fr"
 	"github.com/crate-crypto/go-ipa/banderwagon"
+	"github.com/crate-crypto/go-ipa/common"
 )
 
 func TestPrecompSerde(t *testing.T) {
-	points := GenerateRandomPoints(2)
-	pcl := banderwagon.NewPrecomputeLagrange(points)
-	var buf bytes.Buffer
-
-	err := pcl.SerializePrecomputedLagrange(&buf)
-	if err != nil {
-		t.Fatal(err)
-	}
-	reader := bytes.NewReader(buf.Bytes())
-	deser, err := banderwagon.DeserializePrecomputedLagrange(reader)
-	if err != nil {
-		t.Fatal(err)
+	testNumPoints := []int{
+		2, 3, 7, 20, 64, // Arbitrary sizes
+		common.POLY_DEGREE, // Ethereum Verkle Tries case
 	}
 
-	if !pcl.Equal(*deser) {
-		t.Fatalf("error during (de)serialization of precomputed data %v %v", pcl, deser)
-	}
+	for _, numPoints := range testNumPoints {
+		numPoints := numPoints
+		t.Run(fmt.Sprintf("num points %d", numPoints), func(t *testing.T) {
+			// Generate the random points.
+			points := GenerateRandomPoints(uint64(numPoints))
+
+			// Create thee precomputes.
+			pcl1 := banderwagon.NewPrecomputeLagrange(points)
+
+			// Serialize.
+			var buf bytes.Buffer
+			if err := pcl1.SerializePrecomputedLagrange(&buf); err != nil {
+				t.Fatal(err)
+			}
+
+			// Deserialize and check both are equal.
+			reader := bytes.NewReader(buf.Bytes())
+			pcl2, err := banderwagon.DeserializePrecomputedLagrange(reader)
+			if err != nil {
+				t.Fatal(err)
+			}
 
+			if !pcl1.Equal(*pcl2) {
+				t.Fatalf("error during (de)serialization of precomputed data %v %v", pcl1, pcl2)
+			}
+
+			// Double check that a random evaluation is the same for both too; this should be obvious if .Equal(...)
+			// is correct, but it's good to be sure since it might have a bug.
+			evals := make([]fr.Element, numPoints)
+			for i := 0; i < numPoints; i++ {
+				evals[i].SetRandom()
+			}
+			comm1 := pcl1.Commit(evals)
+			comm2 := pcl2.Commit(evals)
+			if !comm1.Equal(&comm2) {
+				t.Fatalf("random vector commitment between original and deserialized doesn't match: %v %v", comm1, comm2)
+			}
```

Comment:
I added this extra paranoid check. LMK if doesn't make sense.

Reply:
I don't mind leaving this -- it does seem redundant, but if the test times are still okay with it in there, I think its fine